packages/clay-label/src/ClayLabel.tsx

```diff
@@ -46,7 +47,7 @@ const ClayLabel: React.FunctionComponent<Props> = ({
 						className="close"
 						type="button"
 					>
-						{closeButtonProps.children}
+						<Icon symbol="times" />
```

Comment:
As we offer the option to pass **spritemap** to `ClayIcon`, I think it makes sense to add this API to ClayLabel since we are depending on the Icon, if we go without it, we will force the developer to use only the context.

Reply:
Im a bit worried to go down this path of adding a `spritemap` prop to every component that we create that consumes icon. One primary reason it seems awkward is because you would naturally expect `ClayLabel` or any`Clay*` component to use the icons that come from clay-css. One thought is to automatically include the clay icons within `ClayIcon` itself, but thats sort of unclear because of the way people host and consume `clay-css` is different.

Any idea how we could address this? Do you agree its weird to pass spritemap everywhere? My hope was context would fix that problem, but that just means every React app using clay now requires ClayIconContext at the top.

Reply:
One of the problems is that if we do not pass the **spritemap** to non-React applications (e.g Taglibs), I would not be able to use `context` and the only way I see it at the moment would be using the `spritemap` API. Unfortunately I can not think of another way to resolve this outside of that scope.

In taglibs devs do not need to pass the spritemap because we already defined this in the call to the component, but we still need this API, I do not know if we can do something different, in the layers that we will deliver to the Portal we can wrap the components in a type of HOC that uses the context and these layers will have the spritemap API but not ClayLabel for example.

Reply:
yeah thats true about the taglib side. I guess its probably fine to just keep adding spritemap props for now, and then when we get to the other layer to be compatible liferay, we can re-visit it then.

Ill update this PR with spritemap.